### PR TITLE
feat: add remove secret command

### PR DESCRIPTION
**Pull request by yufeiminds**
If someone need update the file list of the `secretGenerator` from directory, they can use `add secret` and `remove secret` together.

Related to #4796

### CODE REVIEW

**Comment by yufeiminds** [comment]
Hi, @natasha41575 @koba1t

Happy to push it forward together. I got a lint issue now.

```shell
kustomize/commands/edit/remove/removesecret_test.go:4:9: package should be `remove_test` instead of `remove` (testpackage)
package remove
        ^
make[1]: *** [lint] Error 1
```

If I want to fix it. Should I make the `newCmdRemoveSecret` function exported?

**Comment by koba1t** [comment]
@yufeiminds 

> If I want to fix it. Should I make the newCmdRemoveSecret function exported?

I think that is probably yes. If you won't find problem, please change that function.

**Comment by yufeiminds** [comment]
@koba1t Lint and test are already passed locally. Could you push it forward again? Thank you~ 😄

**Comment by natasha41575** [comment]
I left some comments on https://github.com/kubernetes-sigs/kustomize/pull/4796. If the same comments apply to here, please make those changes here as well. Thank you!

**Comment by yufeiminds** [comment]
> I left some comments on #4796. If the same comments apply to here, please make those changes here as well. Thank you!

Hi, @natasha41575, @koba1t. I resolved the conversations according to #4796. Please review it again. Thank you~🍻

**Comment by yufeiminds** [comment]
@koba1t Hi, can you review it and push it forward? Thank you. 🍻

**Comment by yufeiminds** [comment]
@natasha41575 Hi, could you help me to push this pull request forward?

File: kustomize/commands/edit/remove/removesecret.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/api/konfig"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kustomize/v4/commands/internal/kustfile"
```
Comment (natasha41575) [inline kustomize/commands/edit/remove/removesecret.go:15]: same comment as https://github.com/kubernetes-sigs/kustomize/pull/4796#discussion_r1313183058

**Review by natasha41575 — APPROVED** [review]
approved, but same comment as https://github.com/kubernetes-sigs/kustomize/pull/4796#issuecomment-1702921615

**Comment by stormqueen1990** [comment]
Closing in favour of #5344 

/close

**Pull request closed without merging**